torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -371,77 +380,61 @@ def _lower_weighted_ref_functional(
-                else:
-                    bias = func_node.kwargs.get("bias", None)
-                prepack_args = (quantized_weight, bias)
-                if ref_func == F.linear:
-                    prepack_op = get_linear_prepack_op_for_dtype(weight_dtype)
-                else:
-                    raise ValueError("Lowering for functional currently only supports linear op")
-                insert_prepack_after = bias if has_bias else quantized_weight
-                with model.graph.inserting_after(insert_prepack_after):
-                    packed_weight = model.graph.create_node("call_function", prepack_op, prepack_args, {})
-
-                # Step 2: Replace reference pattern with the corresponding quantized op
-                func_node.args = (input_dq_node.args[0], packed_weight, output_scale_node, output_zp_node)
-                func_node.target = q_relu_func if is_relu else q_func
-                q_node.replace_all_uses_with(func_node)
-                output_zp_node.append(func_node)
-
-                # Clean up: Remove dequantize and quantize nodes and the old func node
-                for dqn in [input_dq_node, weight_dq_node]:
-                    dqn_input = dqn.args[0]
-                    dqn.replace_all_uses_with(dqn_input)
-                    model.graph.erase_node(dqn)
-                model.graph.erase_node(q_node)
-                if is_relu:
-                    model.graph.erase_node(relu_node)
+        # Step 1: Replace quantized weights with packed weights, which will be folded later
+        # Linear args: (dequantized inputs, dequantized weights[, bias])
+        # Conv args: (dequantized inputs, dequantized weights[, bias, stride, padding, dilation, groups])
+        (input_dq_node, weight_dq_node, *remaining_func_args) = func_node.args
+        quantized_weight = weight_dq_node.args[0]
+        weight_dtype = quantized_weight.args[-1]
+        # Use the right prepack op and prepare the corresponding args
+        prepack_args = tuple([quantized_weight] + remaining_func_args)
+        if func_node.target == F.linear:
+            prepack_op = get_linear_prepack_op_for_dtype(weight_dtype)
+        elif func_node.target in CONV_FUNCTIONAL_OPS:
+            prepack_op = get_qconv_prepack_op(func_node.target)
+        else:
+            raise ValueError("Lowering is not supported for op '%s'" % func_node.target)
+        with model.graph.inserting_before(output_scale_node):
+            packed_weight = model.graph.create_node("call_function", prepack_op, prepack_args, {})
+
+        # Step 2: Replace reference pattern with the corresponding quantized op
+        (q_func, q_relu_func) = LOWER_FUNCTIONAL_MAP[func_node.target]
+        func_node.target = q_relu_func if relu_node is not None else q_func
+        func_node.args = (input_dq_node.args[0], packed_weight, output_scale_node, output_zp_node)
+        q_node.replace_all_uses_with(func_node)
+        output_zp_node.append(func_node)
```

Comment:
what does this mean?

Reply:
This means `func_node` is moved to after `output_zp_node` (actually this line was here before already)

Reply:
maybe add a comment? `append` has the same name as list.append, which is pretty confusing